pkg/reconciler/pipeline/dag/dag.go

```diff
@@ -67,17 +67,16 @@ func (g *Graph) addPipelineTask(t Task) (*Node, error) {
 }
 
 // Build returns a valid pipeline Graph. Returns error if the pipeline is invalid
-func Build(tasks Tasks) (*Graph, error) {
+func Build(tasks Tasks, deps map[string][]string) (*Graph, error) {
```

Comment:
Any reason to pass `deps` there as it comes from `tasks` ? we could just do

```go
func Build(tasks Tasks) (*Graph, error) {
	deps = tasks.Deps()
}
```

Reply:
@vdemeester that's what it currently does - maybe @pritidesai can add more info in the commit message to explain but the reason for this change is to be able to call Build with an empty dep list

in the case of finally tasks, we create 2 DAGs. when finally tasks depend on results from non-finally tasks, Deps() will return the non-finally Tasks (or will it error? im not sure). but in that case we don't want to build connections between the finally tasks and the non-finally tasks

making this change enables us to build a graph in both cases

another option might be to stop treating finally tasks as a graph but i have a strong feeling we will be adding more graph like features in future so im not sure we should do that now

Reply:
oh actually @pritidesai 's commit message does already explain this pretty well i think: https://github.com/tektoncd/pipeline/pull/3583/commits/9716d32b77d5e4a63f7be8c624406dfc28b8e88f

Reply:
oh I did not read the commit :sweat_smile: